packages/network/test/integration/oneway-stream-connections.test.ts

```diff
@@ -0,0 +1,303 @@
+            .toMatchObject(
+                new Error('Joining stream as pure publisher failed on contact-node non-contact-node for stream stream-0#0')
+            )
+    })
+
+    it('Cannot open publish only stream connection to a node without an existing subscription to the given stream', async () => {
+        await expect(onewayNode.joinStreamPartAsPurePublisher(defaultStreamPartId, 'non-contact-node'))
+            .rejects
+            .toMatchObject(
+                new Error('Joining stream as pure publisher failed on contact-node non-contact-node for stream stream-0#0')
+            )
+    })
+
+    it('if caught, failed publish only connections do not clean out existing connections', async () => {
+        await onewayNode.joinStreamPartAsPurePublisher(defaultStreamPartId, 'contact-node')
+        await onewayNode.joinStreamPartAsPurePublisher(defaultStreamPartId, 'contact-node-2')
+        try {
+            await onewayNode.joinStreamPartAsPurePublisher(StreamPartIDUtils.parse('stream-5#0'), 'non-existing-node')
+        } catch (err) {
+        }
+
+        // @ts-expect-error private
+        expect(onewayNode.streamPartManager.getOutboundNodesForStreamPart(defaultStreamPartId)).toContainValues(['contact-node', 'contact-node-2'])
+        // @ts-expect-error private
+        expect(onewayNode.streamPartManager.getNeighborsForStreamPart(defaultStreamPartId)).toBeArrayOfSize(0)
+    })
+
+    it('If publish only connection is the only stream connection on contact node it will not unsubscribe', async () => {
+        await Promise.all([
+            waitForEvent(contactNode, NodeEvent.NODE_UNSUBSCRIBED),
+            contactNode2.unsubscribe(defaultStreamPartId)
+        ])
+        await onewayNode.joinStreamPartAsPurePublisher(defaultStreamPartId, 'contact-node')
+        await Promise.all([
+            waitForEvent(contactNode, NodeEvent.ONE_WAY_CONNECTION_CLOSED),
+            onewayNode.leavePurePublishingStreamPart(defaultStreamPartId, 'contact-node'),
+        ])
+        // @ts-expect-error private
+        expect(contactNode.streamPartManager.isSetUp(defaultStreamPartId)).toBeTrue()
+    })
+
+    it('will reconnect after lost connectivity', async () => {
+        await onewayNode.joinStreamPartAsPurePublisher(defaultStreamPartId, 'contact-node')
+
+        await Promise.all([
+            waitForEvent(onewayNode, NodeEvent.NODE_CONNECTED, 20000),
+            // @ts-expect-error private
+            contactNode.nodeToNode.disconnectFromNode('publisher', 'testing')
+        ])
+
+        await Promise.all([
+            waitForEvent(contactNode, NodeEvent.MESSAGE_RECEIVED),
+            waitForEvent(contactNode2, NodeEvent.MESSAGE_RECEIVED),
+            onewayNode.publish(new StreamMessage({
+                messageId: new MessageID(toStreamID('stream-0'), 0, 120, 0, 'publisher', 'session'),
+                content: {
+                    hello: 'world'
+                },
+            }))
+        ])
+    }, 20000)
```

Comment:
Make this slightly higher than the  `waitForEvent` timeout. This way if the `waitForEvent` timeouts, jest will clearly tell us the line number:
``` suggestion
    }, 20100)
```